modules/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -223,7 +224,9 @@ static AWSCredentialsProvider buildCredentials(
             if (webIdentityTokenCredentialsProvider.isActive()) {
                 logger.debug("Using a custom provider chain of Web Identity Token and instance profile credentials");
                 return new PrivilegedAWSCredentialsProvider(
-                    new AWSCredentialsProviderChain(webIdentityTokenCredentialsProvider, new EC2ContainerCredentialsProviderWrapper())
+                    new CustomAWSCredentialsProviderChain(
```

Comment:
Instead of implementing another `AWSCredentialsProvider` I wonder if we could instead catch and log an error in `CustomWebIdentityTokenCredentialsProvider#getCredentials?`

And rely on `AWSCredentialsProviderChain` as it was before.

Reply:
The catch is that `AWSCredentialsProviderChain` catches errors and silently logs them at the `DEBUG` level,, so we won't get an exception if the first provider fails, but the second succeeds. Hence, we the custom provider that logs individual errors as warnings.

Reply:
I agree and sorry if I wasn't clear. What I'm suggesting is too implement a `FilterAWSCredentialsProvider` that delegates calls to a `AWSCredentialsProvider` but catches and logs at the level we want (I suppose `ERROR`) before rethrowing. 

Someting like:
```java
private static class LoggingCredentialsProvider implements AWSCredentialsProvider {

        private final AWSCredentialsProvider delegate;

        private LoggingCredentialsProvider(AWSCredentialsProvider delegate) {
            this.delegate = Objects.requireNonNull(delegate);
        }

        @Override
        public AWSCredentials getCredentials() {
            try {
                return delegate.getCredentials();
            } catch (Exception e) {
                LOGGER.error("Unable to load credentials from " + delegate, e);
                throw e;
            }
        }

        @Override
        public void refresh() {
            try {
                delegate.refresh();
            } catch (Exception e) {
                LOGGER.error("Unable to refresh " + delegate, e);
                throw e;
            }
            
        }
    }
```

And then just wrap the low-levels providers with it:

```java
logger.debug("Using a custom provider chain of Web Identity Token and instance profile credentials");
                return new AWSCredentialsProviderChain(
                    new LoggingCredentialsProvider(webIdentityTokenCredentialsProvider),
                    new LoggingCredentialsProvider(new EC2ContainerCredentialsProviderWrapper())
                );
```

That would provide the behavior we want without having to reimplement and maintain the logic of AWSCredentialsProviderChain?

Reply:
That's a good option :+1: